File: src/0376-Wiggle-Subsequence/0376.scala
object Solution{
  def wiggleMaxLength(nums:Array[Int]): Int ={
    var _pos, _neg = 1
    for(i <- 1 until nums.length){
      if(nums(i-1) < nums(i)){
        _neg = _pos + 1
      }else if(nums(i-1) < nums(i)) {
        _pos = _neg + 1
      }
    }
    return if (_neg > _pos) _neg else _pos
  }

  def main(args:Array[String]): Unit ={
    val nums = Array(1,17,5,10,13,15,10,5,16,8)
    println(wiggleMaxLength(nums))
  }
}

File: src/0376-Wiggle-Subsequence/0376.py
class Solution:
    def wiggleMaxLength(self, nums):
        size = len(nums)
        _pos, _neg = 1, 1
        for i in range(1, size):
            if nums[i - 1] < nums[i]:
                _neg = _pos + 1
            elif nums[i - 1] > nums[i]:
                _pos = _neg + 1

        return _neg if _pos < _neg else _pos


if __name__ == "__main__":
    nums = [1, 7, 4, 9, 2, 5]
    print Solution().wiggleMaxLength(nums)